TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -295,6 +296,33 @@ public static DataDefinition CheckVariable(Node node, StorageArea storageArea)
         }
     }
 
+    internal class StructureChecker
+    {
+        /// <summary>
+        /// Checks if a structure declaration is valid
+        /// </summary>
+        /// <param name="node"></param>
+        internal static void OnNode(Node node)
+        {
+            //Elements are not checked as they are not a DataDefinitionEntry
+            if (node.CodeElement == null ||
+                !(node.CodeElement is DataDefinitionEntry))
+            {
+                return; 
+            }
+            //Error if node's CodeElement is :
+            //- a rename entry or 
+            //- a data entry that contains a Picture
+            //AND the parent of the element is of type Picture
+            if ((node.CodeElement is DataDescriptionEntry && (node.CodeElement as DataDescriptionEntry).Picture != null) &&
+                (node.Parent != null && node.Parent.CodeElement is DataDescriptionEntry && (node.Parent.CodeElement as DataDescriptionEntry).Picture != null))
```

Comment:
You use `is` followed by a `as`, this means there are 2 cast.
Better use only `as` and check for null. 
This will be faster.